src/main/java/uk/gov/hmcts/reform/roleassignmentbatch/task/DeleteExpiredRecords.java

```diff
@@ -1,155 +1,160 @@
-
-                    public List<RoleAssignmentHistory> extractData(
-                            ResultSet rs) throws SQLException, DataAccessException {
-
-                        List<RoleAssignmentHistory> list = new ArrayList<RoleAssignmentHistory>();
-                        while (rs.next()) {
-                            RoleAssignmentHistory roleAssignmentHistory = new RoleAssignmentHistory();
-                            roleAssignmentHistory.setId(rs.getObject("id", java.util.UUID.class));
-                            roleAssignmentHistory.setRequestId(rs.getObject("request_id", java.util.UUID.class));
-                            roleAssignmentHistory.setActorIDType(rs.getString("actor_id_type"));
-                            roleAssignmentHistory.setActorId(rs.getObject("actor_id", java.util.UUID.class));
-                            roleAssignmentHistory.setRoleType(rs.getString("role_type"));
-                            roleAssignmentHistory.setRoleName(rs.getString("role_name"));
-                            roleAssignmentHistory.setClassification(rs.getString("classification"));
-                            roleAssignmentHistory.setGrantType(rs.getString("grant_type"));
-                            roleAssignmentHistory.setRoleCategory(rs.getString("role_category"));
-                            roleAssignmentHistory.setReadOnly(rs.getBoolean("read_only"));
-                            roleAssignmentHistory.setBeginTime(rs.getTimestamp("begin_time"));
-                            roleAssignmentHistory.setEndTime(rs.getTimestamp("end_time"));
-                            roleAssignmentHistory.setStatus(rs.getString("status"));
-                            roleAssignmentHistory.setReference(rs.getString("reference"));
-                            roleAssignmentHistory.setProcess(rs.getString("process"));
-                            roleAssignmentHistory.setAttributes(rs.getString("attributes"));
-                            roleAssignmentHistory.setNotes(rs.getString("notes"));
-                            roleAssignmentHistory.setLog(rs.getString("log"));
-                            roleAssignmentHistory.setStatusSequence(rs.getInt("status_sequence"));
-                            roleAssignmentHistory.setCreated(rs.getTimestamp("created"));
-                            list.add(roleAssignmentHistory);
-                        }
-                        return list;
-                    }
-                });
-        return rah;
+    public List<RoleAssignmentHistory> getLiveRecordsFromHistoryTable() {
+        String getSQL = "SELECT * from role_assignment_history rah  "
+                        + "WHERE id in (SELECT id FROM role_assignment WHERE end_time <= now()) and status='LIVE'";
+        return
+            getJdbcTemplate().query(getSQL, rs -> {
+                List<RoleAssignmentHistory> list = new ArrayList<>();
+                while (rs.next()) {
```

Comment:
It may throw NullPointer Exception if there is no record set from getSql() result.

Reply:
I tried adding a null check, but IntelliJ is complaining about the same
<img width="620" alt="Screenshot 2020-08-28 at 11 24 34 am" src="https://user-images.githubusercontent.com/49564392/91550904-34f6bf00-e921-11ea-8eb1-962c6ad17765.png">
It seems that null check is not required. 

Checked couple of forums also, they state that the resultset can be empty if ro row is returned. However, it will never be null.